src/app/mapControls/Filter.js

```diff
@@ -0,0 +1,227 @@
+        postCreate: function () {
+            // summary:
+            //      build bubbles
+            console.log('app/mapControls/Filter:postCreate', arguments);
+
+            this.items.forEach(function (item) {
+                var lbl = domConstruct.create('label', {
+                    'class': 'btn btn-default btn-xs',
+                    innerHTML: ' ' + item[0]
+                }, this.buttonContainer);
+                domConstruct.create('input', {
+                    value: item[1],
+                    type: 'checkbox',
+                    autocomplete: 'off',
+                    'onclick': lang.partial(lang.hitch(this, 'itemClicked'), item[1])
+                }, lbl, 'first');
+            }, this);
+
+            if (this.anyAllToggle) {
+                domClass.remove(this.anyAllGroup, 'hidden');
+            }
+
+            $('[data-toggle="tooltip"]', this.domNode).tooltip({
+                delay: {
+                    show: 750,
+                    hide: 100
+                },
+                container: 'body',
+                html: true,
+                placement: 'bottom'
+            });
+            this.inherited(arguments);
+        },
+        clear: function () {
+            // summary:
+            //      unselects all buttons
+            console.log('app/mapControls/Filter:clear', arguments);
+
+            query('.btn', this.buttonContainer).forEach(function (btn) {
+                domClass.remove(btn, 'active');
+                btn.setAttribute('aria-pressed', false);
+            });
+
+            this.numSpan.innerHTML = 0;
+            this.selectedValues = [];
+            this.emit('changed');
+        },
+        itemClicked: function (value) {
+            // summary:
+            //      description
+            // value: String
+            //      value of the item that was clicked
+            console.log('app/mapControls/Filter:itemClicked', arguments);
+
+            var index = this.selectedValues.indexOf(value);
+            if (index === -1) {
+                this.selectedValues.push(value);
+            } else {
+                this.selectedValues.splice(index, 1);
+            }
+
+            this.numSpan.innerHTML = this.selectedValues.length;
+
+            this.emit('changed');
+        },
+        getQuery: function () {
+            // summary:
+            //      assembles all selected values into a def query
+            console.log('app/mapControls/Filter:getQuery', arguments);
+
+            if (this.selectedValues.length) {
+                var values;
+                if (this.fieldType === c.TYPE_TEXT) {
+                    values = this.selectedValues.map(function (v) {
+                        return "'" + v + "'";
+                    });
+                } else {
+                    values = this.selectedValues;
+                }
+                if (this.any) {
+                    var where = this.fieldName + ' IN (' + values.join(', ') + ')';
```

Comment:
i think we can get rid of the space. seems unnecessary unless arcmap needs it. will make the query smaller.